### PR TITLE
fix(worker): fix node typings to pass environement variables correctly when spawning new thread

### PR DESCRIPTION
**Pull request by franklevasseur**
Allright, this fixes a bug posted on the community forum [here](https://forum.botpress.com/t/emulator-not-working-and-unable-to-train/5207/25).

It's probably to unluckiest bug I've seen in my short dev career...

Basically what happened is:

- In file `packages/worker/src/thread-pool.ts` at line `13`
- I was passing env variables as the option object: `new Thread(entryPoint, { ...env })`
- The correct way to do is to set the field for env: `new Thread(entryPoint, { env: myEnvObject })`
- However this did not build, because the worker package was referencing node typings of an old version (v11.x.x)
- This was not harmful in most cases as the `options.env` was simply undefined which defaults to `process.env`
- However when using [pm2](https://pm2.keymetrics.io/) the environment variable `'env'` is defined with value `"[object Object]"`... (THIS IS SO UNLUCKY)

This resulted in the following error:
```
The "options.env" property must be of type object or one of undefined, null, or worker_threads.SHARE_ENV. Received type string ('[object Object]')
```

Well, its fixed now.

Until its release, please consider not using pm2.

### CODE REVIEW

File: packages/worker/src/thread-pool.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export class ThreadPool<I, O> extends WorkerPool<I, O> {
   }
 
   createWorker = async (entryPoint: string, env: NodeJS.ProcessEnv) => {
-    const worker = new Thread(entryPoint, { ...env })
+    const worker = new Thread(entryPoint, { env })
```
Comment (franklevasseur) [inline packages/worker/src/thread-pool.ts:13]: This is the fix

File: packages/worker/package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
   },
   "devDependencies": {
     "@types/lodash": "^4.14.116",
-    "@types/yn": "^3.1.0"
+    "@types/yn": "^3.1.0",
+    "@types/node": "^12.13.0"
```
Comment (franklevasseur) [inline packages/worker/package.json:17]: This makes sur package `@botpress/worker` uses the correct node typings

File: packages/worker/tsconfig.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
     "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
     "types": ["node"],
     "baseUrl": ".",
-    "composite": true
+    "composite": true,
+    "typeRoots": ["./node_modules/@types", "../../node_modules/@types"]
```
Comment (franklevasseur) [inline packages/worker/tsconfig.json:11]: This makes sur package `@botpress/worker` uses the correct node typings

**Review by EFF — APPROVED** [review]
Well done !

**Pull request merged**